tests/check_framework/test_security.py

```diff
@@ -534,3 +534,18 @@ def test_allowed_hosts_empty(self):
     @override_settings(ALLOWED_HOSTS=['.example.com'])
     def test_allowed_hosts_set(self):
         self.assertEqual(self.func(None), [])
+
+
+class CheckTypeOfAllowedHostsTest(SimpleTestCase):
+    @property
+    def func(self):
+        from django.core.checks.security.base import check_type_of_allowed_hosts
+        return check_type_of_allowed_hosts
+
+    @override_settings(ALLOWED_HOSTS='example.com')
+    def test_allowed_hosts_is_string(self):
+        self.assertEqual(self.func(None), [base.W022])
```

Comment:
I see the other tests are doing it but, I don't see why we're declaring `func` as an `@property` here and/or passing `None` when calling it. (Would not a normal method do? Why pass any argument at all? `self` would be fine, but is unused.)

Reply:
`None` is being passed for the `app_configs` argument - even though the check ignores `app_configs`, `[]` should be used as a more valid test value.

I think it's worth looking into refactoring these tests in general, as you're write these property inner-imports are a bit confusing, made a note to self.

Reply:
yeah I don't see a reason why the imports are not top-level and we don't call the functions directly.

Reply:
It's the pattern that django-secure used. Not sure if the reason is still relevant. \cc @carljm